src/index.ts

```diff
@@ -484,6 +484,11 @@ async function startServer(
 	port: string | number,
 ): Promise<Server> {
 	let server: Server;
+	// empty 404 error for undefined paths
+	app.use((_request, response) => {
```

Comment:
Just asking: Given that we already support tons of options in the setup call (eg onInit, onInitRoutes, ...) and ppl should be able to already register everything using those already, would it make more sense if we did this as he last step of `setup()`, rather than in the `startServer()`?